exercises/takudzwa_makoni/ex02_oo_basics/src/rectangle.h

```diff
@@ -0,0 +1,29 @@
+#ifndef RECTANGLE_H
+#define RECTANGLE_H
+
+class rectangle : public shape {
+
+private:
+
+	
+	const std::string name;
+	const int numOfSides;
```

Comment:
Can your rectangle have a negative number of sides? I would recommend using `unsigned int` for these cases (see other shape subtypes with similar problems too).